source/gfoidl.Stochastics/Statistics/Sample.AddRange.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using gfoidl.Stochastics.Builders;
+
+namespace gfoidl.Stochastics.Statistics
+{
+    partial class Sample
+    {
+        public void AddRange(IEnumerable<double> values)
+        {
+            if (values == null) ThrowHelper.ThrowArgumentNull(nameof(values));
+
+            if (values is double[] array)
+            {
+                _values = array;
+                return;
+            }
+
+            double min       = double.MaxValue;
+            double max       = double.MinValue;
+            double avg       = 0;
+            int count        = 0;
+            var arrayBuilder = new ArrayBuilder<double>(true);
+
+            using (IEnumerator<double> enumerator = values.GetEnumerator())
+            {
+                while (enumerator.MoveNext())
```

Comment:
This is exactly the same as `foreach`, so use foreach because it's more readable.